### PR TITLE
[2.0] Updates

### PR DESCRIPTION
**Pull request by aman-goel**
Updates include:
- Programmatically set the dependency version in the generated `<project-name>.csproj`
- Minor corrections to PSym (relating to union-find with dynamic machine creation)

### CODE REVIEW

File: Src/PCompiler/CompilerCore/Backend/CSharp/CSharpCodeGenerator.cs [hunk 1]
```diff
@@ -46,8 +46,16 @@ public void Compile(ICompilerConfiguration job)
                     }
                     foreignInclude += "  </ItemGroup>";
                 }
-
                 csprojTemplate = csprojTemplate.Replace("-foreign-include-", foreignInclude);
+
```
Comment (ankushdesai) [inline Src/PCompiler/CompilerCore/Backend/CSharp/CSharpCodeGenerator.cs:50]: Nice! I like it!

File: Src/PRuntimes/PSymRuntime/pom.xml [hunk 2]
```diff
@@ -428,7 +428,7 @@
     <maven.compiler.target>11</maven.compiler.target>
     <log4j2.configurationFile>${project.basedir}/src/main/resources/log4j2.xml</log4j2.configurationFile>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-    <revision>1.0.0</revision>
+    <revision>1.0.1</revision>
```
Comment (ankushdesai) [inline Src/PRuntimes/PSymRuntime/pom.xml:431]: Do we want to make this versioning consistent with the P versions? Just like we do it for all the other P nuget packages?

File: Src/PRuntimes/PSymRuntime/src/main/java/psym/valuesummary/util/ValueSummaryUnionFind.java [hunk 3]
```diff
@@ -32,9 +32,30 @@ public Map<Set<PrimitiveVS>, Guard> getLastUniverseMap() {
         for (Set<PrimitiveVS> set : lastDisjointSet) {
             lastUniverseMap.put(set, universe.get(find(set.iterator().next())));
         }
+        assert(sanityCheck(lastUniverseMap));
         return lastUniverseMap;
     }
 
+    public boolean sanityCheck(Map<Set<PrimitiveVS>, Guard> universeMap) {
```
Comment (ankushdesai) [inline Src/PRuntimes/PSymRuntime/src/main/java/psym/valuesummary/util/ValueSummaryUnionFind.java:39]: Yo! is this a bug fix?

File: Src/PRuntimes/PSymRuntime/pom.xml (hunk 2 above)
Comment (aman-goel) [reply Src/PRuntimes/PSymRuntime/pom.xml:431]: Noted! We will do this when we make PSym release.

File: Src/PRuntimes/PSymRuntime/src/main/java/psym/valuesummary/util/ValueSummaryUnionFind.java (hunk 3 above)
Comment (aman-goel) [reply Src/PRuntimes/PSymRuntime/src/main/java/psym/valuesummary/util/ValueSummaryUnionFind.java:39]: Yes. When dynamic machine creation is involved, [ValueSummaryUnionFind](https://github.com/p-org/P/blob/1fda3058ca3b0bcd28efe494d3d6528bb12a3052/Src/PRuntimes/PSymRuntime/src/main/java/psym/valuesummary/util/ValueSummaryUnionFind.java#L12) breaks and was no longer producing disjoint sets of guards. I corrected this and added this sanity check.

**Review by ankushdesai — APPROVED** [review]

**Pull request merged**